### PR TITLE
Fixes related to text formatting

### PR DESCRIPTION
**Pull request by JoshyPHP**
* smiley_text() is specific to the 3.1 formatting and shouldn't be used on 3.2.
* enable_bbcode() and the other functions described are part of the parser service's API.
* get_parser() retrieves the underlying s9e\TextFormatter\Parser instance, which has a different API.

### CODE REVIEW

**Comment by VSEphpbb** [comment]
Why did you remove smiley_text()? It's still available in 3.2 and can still be used as it behaved in 3.1, which is why it was mentioned.

File: development/extensions/new_in_rhea.rst [hunk 1]
```diff
@@ -98,15 +98,13 @@ Fortunately, the integration is pretty seamless and most existing extensions tha
   * ``generate_text_for_edit()``
   * ``generate_text_for_storage()``
   * ``strip_bbcode()``
-  * ``smiley_text()``
```
Comment (VSEphpbb) [inline development/extensions/new_in_rhea.rst]: Why did you remove smiley_text()? It's still available in 3.2 and can still be used as it behaved in 3.1.

Comment (JoshyPHP) [inline development/extensions/new_in_rhea.rst]: Because the thing says "For example, the following phpBB functions will continue to work as they did in phpBB 3.1" which is misleading. People may think that `smiley_text()` can be used on 3.2 content. The other functions in that list can, but this one only works on 3.1.

Comment (VSEphpbb) [inline development/extensions/new_in_rhea.rst]: Why can't it? I'm using smiley_text() in 3.2 the same as in 3.1 (to unrender smileys back to text) and it seems to work.

Comment (VSEphpbb) [inline development/extensions/new_in_rhea.rst]: I see what you're saying.

FYI if you use its force_option it did make smilies appear in their text form instead of the image form:
https://github.com/phpbb/phpbb/blob/master/phpBB/includes/functions_content.php#L1040-L1043

It has not been deprecated for B.C. reasons. But I see that it has no use for 3.2 parsed text content, so I guess it is OK to remove from here.
